### PR TITLE
Blood magic fixes

### PR DESCRIPTION
**Pull request by ghost**
:cl:
fix: Twisted Construction no longer damages the caster if canceled or failed to cast
fix: Twisted Construction's tool-tip now correctly specifies the amount of metal needed for a construct shell
refactor: Blood rite and Twisted Construction code now uses defines instead of magic numbers
/:cl:

### CODE REVIEW

**Comment by py01** [comment]
There are still some magic numbers and unclear tooltips, which i am going to fix in a separate PR changing only desc and /examine() code for blood magic

**Comment by adamkad1** [comment]
now maybe let manifest ghosters move off the rune because some ghosts are too think to listen to guys screaming they are dying

**Comment by py01** [comment]
I'm not familiar with appveyor, don't know what the error means.

File: code/modules/antagonists/cult/blood_magic.dm [hunk 1]
```diff
@@ -549,7 +549,7 @@
 	. = ..()
 
 
-//Construction: Creates a construct shell out of 50 metal sheets, or converts plasteel into runed metal
+//Construction: Converts 50 metal to a construct shell, plasteel to runed metal, airlock to brittle runed airlock, a borg to a construct, or borg shell to construct shell
```
Comment (ATH1909) [inline code/modules/antagonists/cult/blood_magic.dm]: "a borg to a construct, or borg shell to _**a**_ construct shell"

File: code/modules/antagonists/cult/blood_magic.dm [hunk 2]
```diff
@@ -762,10 +766,10 @@
 			return
 		switch(choice)
 			if("Blood Spear (150)")
```
Comment (Cyberboss) [inline code/modules/antagonists/cult/blood_magic.dm]:
```suggestion
			if("Blood Spear ([BLOOD_SPEAR_COST])")
```

File: code/modules/antagonists/cult/blood_magic.dm [hunk 3]
```diff
@@ -778,23 +782,23 @@
...
 					var/obj/rite = new /obj/item/gun/ballistic/shotgun/boltaction/enchanted/arcane_barrage/blood()
-					uses -= 300
+					uses -= BLOOD_BARRAGE_COST
 					qdel(src)
 					if(user.put_in_hands(rite))
 						to_chat(user, "<span class='cult'><b>Your hands glow with power!</b></span>")
 					else
 						to_chat(user, "<span class='cultitalic'>You need a free hand for this rite!</span>")
 						qdel(rite)
 			if("Blood Beam (500)")
```
Comment (Cyberboss) [inline code/modules/antagonists/cult/blood_magic.dm]:
```suggestion
			if("Blood Beam ([BLOOD_BEAM_COST])")
```

File: code/modules/antagonists/cult/blood_magic.dm [hunk 4]
```diff
@@ -778,23 +782,23 @@
 						user.visible_message("<span class='warning'>A [rite.name] appears at [user]'s feet!</span>", \
 							 "<span class='cultitalic'>A [rite.name] materializes at your feet.</span>")
 			if("Blood Bolt Barrage (300)")
```
Comment (Cyberboss) [inline code/modules/antagonists/cult/blood_magic.dm]:
```suggestion
			if("Blood Bolt Barrage ([BLOOD_BARRAGE_COST])")
```

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/antagonists/cult/blood_magic.dm [hunk 5]
```diff
@@ -176,7 +176,7 @@
 
 /datum/action/innate/cult/blood_spell/construction
 	name = "Twisted Construction"
-	desc = "<u>A sinister spell used to convert:</u><br>Plasteel into runed metal<br>25 metal into a construct shell<br>Cyborgs directly into constructs<br>Cyborg shells into construct shells<br>Airlocks into runed airlocks (harm intent)"
+	desc = "<u>A sinister spell used to convert:</u><br>Plasteel into runed metal<br>50 metal into a construct shell<br>Cyborgs directly into constructs<br>Cyborg shells into construct shells<br>Airlocks into runed airlocks (harm intent)"
```
Comment (Cyberboss) [inline code/modules/antagonists/cult/blood_magic.dm]:
```suggestion
	desc = "<u>A sinister spell used to convert:</u><br>Plasteel into runed metal<br>[METAL_TO_CONSTRUCT_SHELL_CONVERSION] metal into a construct shell<br>Cyborgs directly into constructs<br>Cyborg shells into construct shells<br>Airlocks into runed airlocks (harm intent)"
```
Not sure if that will compile though

Comment (py01) [reply code/modules/antagonists/cult/blood_magic.dm]: I've tried putting variables into desc, doesn't compile. I am planning to put that info with a variable into a /examine() in a seperate pr.

File: code/modules/antagonists/cult/blood_magic.dm (hunk 4 above)
Comment (py01) [reply code/modules/antagonists/cult/blood_magic.dm]: List options do not allow variables, only static strings it seems.

Comment (py01) [reply code/modules/antagonists/cult/blood_magic.dm]: Planning on moving costs to the tooltip with /examine() in a seperate PR

**Comment by py01** [comment]
Not sure what else needs to be done before merging.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**